src/ng-select/lib/selection-model.ts

```diff
@@ -62,14 +64,25 @@ export class DefaultSelectionModel implements SelectionModel {
     }
 
     private _setChildrenSelectedState(children: NgOption[], selected: boolean) {
-        children.forEach(x => x.selected = selected);
+        children.forEach(x => {
+            if (x.disabled !== true) {
+                x.selected = selected;
```

Comment:
I don't think this works for all cases. Consider the following:

```
children = [
  { val: 1, selected: false },
  { val: 2, selected: true, disabled: true },
  { val: 3, selected: false, disabled: false }
]

_setChildrenSelectedState(children, false);

// Expected: selected children = [2]
// Actual: selected children = []

Reply:
what about 
```js
x.selected = x.disabled || selected;
```
or 
```js
if (x.disabled) { 
continue;
}
x.selected = selected
```

Reply:
@NicolasPennie this is the last improvement to make it more readable, everything else looks good. Thank you for your contribution! 🎉

Reply:
First one won't work since it will select a disabled item. The second option works fine but I need to switch it to a for..of since you can't call `continue` within `Array.prototype.forEach`.

Reply:
Updated! 👍